assets/js/base/components/payment-methods/payment-methods.js

```diff
@@ -81,16 +86,38 @@ const PaymentMethods = () => {
 				currentPaymentMethods.current,
 				isEditor
 			);
+			const { supports } = currentPaymentMethods.current[
+				activePaymentMethod
+			];
 			return paymentMethod ? (
 				<PaymentMethodErrorBoundary isEditor={ isEditor }>
 					{ cloneElement( paymentMethod, {
 						activePaymentMethod,
 						...currentPaymentMethodInterface.current,
 					} ) }
+					{ customerId > 0 && supports.savePaymentInfo && (
+						<CheckboxControl
+							className="wc-block-checkout__save-card-info"
+							label={ __(
+								'Save payment information to my account for future purchases.',
+								'woo-gutenberg-products-block'
+							) }
+							checked={ shouldSavePayment }
+							onChange={ () =>
+								setShouldSavePayment( ! shouldSavePayment )
```

Comment:
This is minor, but this seems risky to me since it toggles the boolean without considering the state of `checked`. I know in this context we know the initial state but I'm wondering what would happen if something modified this firing events.

Reply:
> I'm wondering what would happen if something modified this firing events.

Do you have some examples in mind where that might occur? Are you talking about something hooking directly into the dom event?

Reply:
Regular javascript, possibly browser autocomplete (if that doesn't fire change events?). Basically this code is trusting the initial state without checking the state onChange.

Reply:
🤔 I think what we're doing here is basically saying the value of the checkbox is connected to the current state. So thinking through what _could_ happen:

- If external javascript changes the checkbox, it's not going to do anything (because it depends on the app state). IMO that's okay, this is a controlled component - we don't _want_ external javascript affecting the app state.
- If autocomplete causes state to change... what autocomplete would that be? This isn't a standard autocomplete field?

By setting state this way we're essentially saying that the _state_ of the checkbox is controlled by the component and _reacts_ to user changes on the checkbox.